pulpcore/cli/python/repository.py

```diff
@@ -0,0 +1,166 @@
+    PulpContext,
+    PulpEntityContext,
+    PulpRepositoryContext,
+    pass_pulp_context,
+    pass_repository_context,
+)
+from pulpcore.cli.common.generic import (
+    create_command,
+    destroy_command,
+    href_option,
+    label_command,
+    label_select_option,
+    list_command,
+    name_option,
+    show_command,
+    update_command,
+    version_command,
+)
+from pulpcore.cli.python.context import (
+    PulpPythonContentContext,
+    PulpPythonRemoteContext,
+    PulpPythonRepositoryContext,
+)
+
+
+def _remote_callback(
+    ctx: click.Context, param: click.Parameter, value: Optional[str]
+) -> Optional[Union[str, PulpEntityContext]]:
+    # Pass None and "" verbatim
+    if value:
+        pulp_ctx: PulpContext = ctx.find_object(PulpContext)
+        return PulpPythonRemoteContext(pulp_ctx, entity={"name": value})
+    return value
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "repo_type",
+    type=click.Choice(["python"], case_sensitive=False),
+    default="python",
+)
+@pass_pulp_context
+@click.pass_context
+def repository(ctx: click.Context, pulp_ctx: PulpContext, repo_type: str) -> None:
+    if repo_type == "python":
+        ctx.obj = PulpPythonRepositoryContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+lookup_options = [href_option, name_option]
+create_options = [
+    click.option("--name", required=True),
+    click.option("--description"),
+    click.option("--remote", callback=_remote_callback),
+]
+update_options = [
+    click.option("--description"),
+    click.option("--remote", callback=_remote_callback),
+]
+
+repository.add_command(list_command(decorators=[label_select_option]))
+repository.add_command(show_command(decorators=lookup_options))
+repository.add_command(destroy_command(decorators=lookup_options))
+repository.add_command(create_command(decorators=create_options))
+repository.add_command(update_command(decorators=lookup_options + update_options))
+repository.add_command(version_command())
+repository.add_command(label_command())
+
+
+@repository.command()
+@name_option
+@href_option
+@click.option("--remote")
+@pass_repository_context
+@pass_pulp_context
+def sync(
+    pulp_ctx: PulpContext,
+    repository_ctx: PulpRepositoryContext,
+    remote: Optional[str],
+) -> None:
+    repository = repository_ctx.entity
+    repository_href = repository_ctx.pulp_href
+
+    body = {}
+
+    if remote:
+        remote_href: str = PulpPythonRemoteContext(pulp_ctx, entity={"name": remote}).pulp_href
+        body["remote"] = remote_href
+    elif repository["remote"] is None:
+        name = repository["name"]
+        raise click.ClickException(
+            f"Repository '{name}' does not have a default remote. Please specify with '--remote'."
+        )
+
+    repository_ctx.sync(
+        href=repository_href,
+        body=body,
+    )
+
+
+@repository.command()
+@name_option
+@href_option
+@click.option("--relative-path", required=True)
```

Comment:
Would it be nicer to call this "filename"? (It seems to be "filename" on the content model.)